tools/single_nuc.py

```diff
@@ -0,0 +1,669 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import re
+
+#-------------------- Helper Functions --------------------
+
+def select_sublist(x, y):
+    '''
+    Select intersection of list x and list y. Additionally, capture indices
+        of said intersection on list x.
+
+    Arguments:
+        x (list): Reference list.
+        y (list): Smaller list sharing some or all elements with x.
+
+    Returns:
+        sublist (list): Intersection of x and y.
+        indices (list): Index or indices of intersection on x.
+    '''
+
+
+    sublist = [
+        val for i, val in enumerate(x, start=1)
+        if val in y or i in (y if isinstance(y, list) else [y])
+        ]
+    indices = [i for i, val in enumerate(x) if val in sublist]
```

Comment:
It might be simpler to use `list.index()` for this: https://stackoverflow.com/questions/176918/how-can-i-find-the-index-for-a-given-item-in-a-list